tinygrad/runtime/ops_clang.py

```diff
@@ -9,15 +9,24 @@
   'Darwin': {'cflags':'-lm -fPIC --rtlib=compiler-rt ', 'ext':'dylib', 'exp':''}
 }[platform.system()]
 
+def _compile_to_lib(prg:str):
+  # TODO: is there a way to not write this to disk?
+  dst = f"{tempfile.gettempdir()}/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
+  if not os.path.exists(dst):
+    subprocess.check_output(args=('clang -shared -O2 -Wall -Werror -march=native -x c '+args['cflags']+' - -o '+dst+'.tmp').split(), input=prg.encode('utf-8'))
+    os.rename(dst+'.tmp', dst)
+  return ctypes.CDLL(dst)
+
+if platform.system() == 'Linux':
+  # Reproducer taken from https://github.com/llvm/llvm-project/issues/56204
+  lib = _compile_to_lib('float __attribute__((noinline)) n(float v) { return -v; } int check() { return (__fp16)n(-1.f); }')
+  if lib['check']() != 1:
+    raise RuntimeError('Test lib incorrectly handles half precision floating point. CLANG backend won\'t produce valid results')
+
 class ClangProgram:
   def __init__(self, name:str, prg:str):
-    prg = '#include <math.h>\n#define max(x,y) ((x>y)?x:y)\n#define int64 long\n#define half __fp16\n#define uchar unsigned char\n#define bool uchar\n' + prg
-    # TODO: is there a way to not write this to disk?
-    fn = f"{tempfile.gettempdir()}/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
-    if not os.path.exists(fn):
-      subprocess.check_output(args=('clang -shared -O2 -Wall -Werror -x c '+args['cflags']+' - -o '+fn+'.tmp').split(), input=prg.encode('utf-8'))
-      os.rename(fn+'.tmp', fn)
-    self.lib = ctypes.CDLL(fn)
+    prg = '#include <math.h>\n#define max(x,y) ((x>y)?x:y)\n#define int64 long long\n#define half __fp16\n#define uchar unsigned char\n#define bool uchar\n' + prg
```

Comment:
I would say revert this change for now until we can get windows+clang CI fixed, progress in #1375.

Reply:
Ah good shout. I'll drop the commit now that it's been picked over so a merge doesn't accidentally revert the pick.